src/main/java/com/enonic/xp/app/welcome/WelcomePageScriptBean.java

```diff
@@ -465,7 +466,12 @@ private List<String> getApplicationAdminToolsUris( final ApplicationKey applicat
         AdminToolDescriptors descriptors = service.getByApplication( applicationKey );
 
         return descriptors.getList().stream().map(
-            descriptor -> service.generateAdminToolUri( applicationKey.toString(), descriptor.getName() ) ).collect( Collectors.toList() );
+            descriptor -> makeAdminToolUri( applicationKey, descriptor ) ).collect( Collectors.toList() );
+    }
+
+    private String makeAdminToolUri( final ApplicationKey applicationKey, final AdminToolDescriptor descriptor )
+    {
+        return "admin/" + applicationKey + "/" + descriptor.getName();
```

Comment:
```suggestion
        return "/admin/" + applicationKey + "/" + descriptor.getName();
```
welcome is always on root., so it is not a big deal, but server-relative urls are a bit more easy to deal with